src/module/dialogs/combat/CombatDefenseDialog.js

```diff
@@ -225,172 +440,293 @@ export class CombatDefenseDialog extends FormApplication {
       this.render();
     });
 
-    html.find('.send-mystic-defense').click(() => {
-      const { modifier, spellUsed, spellGrade } = this.modalData.defender.mystic;
-      const { at } = this.modalData.defender.combat;
-
-      if (spellUsed) {
-        const magicProjection =
-          this.defenderActor.system.mystic.magicProjection.imbalance.defensive.final
-            .value;
-        const baseMagicProjection =
-          this.defenderActor.system.mystic.magicProjection.imbalance.defensive.base.value;
-
-        let formula = `1d100xa + ${magicProjection} + ${modifier ?? 0}`;
-        if (this.modalData.defender.withoutRoll) {
-          // Remove the dice from the formula
-          formula = formula.replace('1d100xa', '0');
+    html.find('.send-mystic-defense').click(async () => {
+      const {
+        mystic: {
+          magicProjection,
+          modifier,
+          spellUsed,
+          spellGrade,
+          spellCasting,
+          supernaturalShield: { shieldUsed, newShield }
+        },
+        combat: { at },
+        blindnessPen
+      } = this.modalData.defender;
+      const { i18n } = game;
+      const { spells } = this.defenderActor.system.mystic;
+      const { supernaturalShields } = this.defenderActor.system.combat;
+      let spell,
+        supShield = { create: false };
+      const defenderCombatMod = {
+        modifier: { value: modifier, apply: true },
+        blindnessPen: { value: blindnessPen, apply: true }
+      };
+
+      if (!newShield) {
+        if (!shieldUsed) {
+          return ui.notifications.warn(
+            i18n.localize('macros.combat.dialog.warning.supernaturalShield.mystic')
+          );
         }
-        if (baseMagicProjection >= 200) {
-          // Mastery reduces the fumble range
-          formula = formula.replace('xa', 'xamastery');
+        spell = supernaturalShields.find(w => w._id === shieldUsed);
+        supShield = { ...spell, create: false, id: shieldUsed };
```

Comment:
En realidad, `spell` es un escudo sobrenatural, por lo que el nombre resulta confuso. Quizás es mejor algo tipo
```suggestion
        supShield = supernaturalShields.find(w => w._id === shieldUsed);
        supShield.create = false;
```
Se usa el `.create` para algo?

Reply:
Si